### PR TITLE
SRCH-3045: remove docker files and update readme to use search-services

### PR DESCRIPTION
**Pull request by krbhavith**
## Summary
- Removed docker and env files
- Updated secrets.yml.dev file with environment variables
- Updated Readme according to the changes (docker usage steps)
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review:

#### Functionality Checks

- [x] Code is functional.

- [x] Automated checks pass, if applicable. If Code Climate checks do not pass, explain reason for failures:

- [x] If your changes will be tested manually, you have run `bundle update` and committed your changes to Gemfile.lock.
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
 
- [x] If your target branch is NOT `main`, specify the reason:
 
- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release #.#.#** matching the release number
 
- [x] You have squashed your commits into a single commit (exception: your PR includes commits with formatting-only changes, such as required by Rubocop)

- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket
 
#### Process Checks

- [x] You have specified an "Assignee", and if necessary, additional reviewers

### CODE REVIEW

**Comment by amanifarooque** [comment]
@krbhavith  - could you set the assignee to Martha? The assignee should represent who is testing the PR, so that it shows up in their PR queue. Thank you!

**Review by MothOnMars — CHANGES_REQUESTED** [review]
See comments. We should also make the Kibana instructions version-agnostic:
```
* [Kibana](https://www.elastic.co/kibana) - Kibana is not required, but can be very useful for debugging Elasticsearch. Confirm Kibana is available by visiting <http://localhost:5601>. 
```

File: README.md [hunk 1]
```diff
@@ -14,26 +14,16 @@ Read our [contributing guidelines](https://github.com/GSA/search-gov/blob/master
...
+The required services (MySQL & Elasticsearch) can be run using Docker. Please refer to [searchgov-services](https://github.com/GSA/search-services) for detailed instructions on centralized configuration for services.
 
 * [Elasticsearch](https://www.elastic.co/elasticsearch/) 6.8 - for full-text search and query analytics
 
-We have configured Elasticsearch 6.8 to run on the default port [9200](http://localhost:9200/), and Elasticsearch 7.8 to run on [9278](http://localhost:9278/) for development purposes. To check Elasticsearch settings and directory locations:
+We have configured Elasticsearch 6.8 to run on the default port [9200](http://localhost:9200/), and Elasticsearch 7.8 to run on [9207](http://localhost:9207/) for development purposes. To use a different host (with or without port) or set of hosts, set the `ES_HOSTS` environmental variable. To check Elasticsearch settings and directory locations:
 
     $ curl "localhost:9200/_nodes/settings?pretty=true"
-    $ curl "localhost:9278/_nodes/settings?pretty=true"
+    $ curl "localhost:9207/_nodes/settings?pretty=true"
```
[the remark this answers is not in the document]
Comment (krbhavith) [reply README.md]: removed commands as those are not needed

File: README.md [hunk 2]
```diff
@@ -14,26 +14,16 @@ Read our [contributing guidelines](https://github.com/GSA/search-gov/blob/master
...
-    
-Alternatively, run the services individually, i.e.:
-
-    $ docker compose up redis
+The required services (MySQL & Elasticsearch) can be run using Docker. Please refer to [searchgov-services](https://github.com/GSA/search-services) for detailed instructions on centralized configuration for services.
 
 * [Elasticsearch](https://www.elastic.co/elasticsearch/) 6.8 - for full-text search and query analytics
 
-We have configured Elasticsearch 6.8 to run on the default port [9200](http://localhost:9200/), and Elasticsearch 7.8 to run on [9278](http://localhost:9278/) for development purposes. To check Elasticsearch settings and directory locations:
+We have configured Elasticsearch 6.8 to run on the default port [9200](http://localhost:9200/), and Elasticsearch 7.8 to run on [9207](http://localhost:9207/) for development purposes. To use a different host (with or without port) or set of hosts, set the `ES_HOSTS` environmental variable. To check Elasticsearch settings and directory locations:
```
[the remark this answers is not in the document]
Comment (krbhavith) [reply README.md]: removed version number to keep it generic

File: README.md [hunk 3]
```diff
@@ -14,26 +14,16 @@ Read our [contributing guidelines](https://github.com/GSA/search-gov/blob/master
...
-### Services
-
-All the required services below can be run using [Docker Compose](https://docs.docker.com/compose/):
-
-    $ docker compose up
-    
-Alternatively, run the services individually, i.e.:
-
-    $ docker compose up redis
+The required services (MySQL & Elasticsearch) can be run using Docker. Please refer to [searchgov-services](https://github.com/GSA/search-services) for detailed instructions on centralized configuration for services.
```
[the remark this answers is not in the document]
Comment (krbhavith) [reply README.md]: Make sense, as we are using other services too. I will tweak the sentence.

File: README.md [hunk 4]
```diff
@@ -16,19 +16,18 @@ Use [RVM](https://rvm.io/) to install the version of Ruby specified in [.ruby-ve
...
+    ES_HOSTS=localhost:9207 bin/rspec spec
 
-    $ curl "localhost:9200/_nodes/settings?pretty=true"
-    $ curl "localhost:9207/_nodes/settings?pretty=true"
-    
 Some specs depend upon Elasticsearch having a valid trial license. A 30-day trial license is automatically applied when the cluster is initially created. If your license expires, you can rebuild the cluster by [rebuilding the container and its data volume](https://github.com/GSA/search-gov/wiki/Docker-Command-Reference/_edit#recreate-an-elasticsearch-cluster-useful-for-restarting-a-trial-license). 
     
     
-* [Kibana](https://www.elastic.co/kibana) - Kibana is not required, but can be very useful for debugging Elasticsearch. Confirm Kibana is available for the Elasticsearch 6.8 cluster by visiting <http://localhost:5601>. Kibana for the Elasticsearch 7 cluster should be available on <http://localhost:5678>.
+* [Kibana](https://www.elastic.co/kibana) - Kibana is not required, but can be very useful for debugging Elasticsearch. Confirm Kibana is available for all running versions of Elasticsearch clusters by visiting <http://localhost:5601> and <http://localhost:5678>.
```
Comment (MothOnMars) [inline README.md:30]: See https://github.com/GSA/search-gov/pull/935#pullrequestreview-987603835

File: README.md [hunk 5]
```diff
@@ -27,7 +27,7 @@ We have configured Elasticsearch to run on the default port [9200](http://localh
 Some specs depend upon Elasticsearch having a valid trial license. A 30-day trial license is automatically applied when the cluster is initially created. If your license expires, you can rebuild the cluster by [rebuilding the container and its data volume](https://github.com/GSA/search-gov/wiki/Docker-Command-Reference/_edit#recreate-an-elasticsearch-cluster-useful-for-restarting-a-trial-license). 
     
     
-* [Kibana](https://www.elastic.co/kibana) - Kibana is not required, but can be very useful for debugging Elasticsearch. Confirm Kibana is available for all running versions of Elasticsearch clusters by visiting <http://localhost:5601> and <http://localhost:5678>.
+* [Kibana](https://www.elastic.co/kibana) - Kibana is not required, but can be very useful for debugging Elasticsearch. Confirm Kibana is available for Elasticsearch cluster by visiting <http://localhost:5601>.
```
Comment (MothOnMars) [inline README.md:30]: Please use the recommended, simplified language:
```
* [Kibana](https://www.elastic.co/kibana) - Kibana is not required, but can be very useful for debugging Elasticsearch. Confirm Kibana is available by visiting <http://localhost:5601>. 
```

**Review by MothOnMars — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (krbhavith) [reply README.md:30]: Oh, you already gave the exact sentence, my bad I used my own language.

File: README.md [hunk 6]
```diff
@@ -6,39 +6,28 @@
  [![Maintainability](https://api.codeclimate.com/v1/badges/fd0577360749c9b3d166/maintainability)](https://codeclimate.com/github/GSA/search-gov/maintainability)
 
 ## Contributing to search-gov
-Read our [contributing guidelines](./CONTRIBUTING.md). 
+Read our [contributing guidelines](https://github.com/GSA/search-gov/blob/master/CONTRIBUTING.md).
```
Comment (MothOnMars) [inline README.md:9]: Looks like the merge commit pulled in an unintended change. This should be a relative link:
```
Read our [contributing guidelines](./CONTRIBUTING.md). 
```

**Review by MothOnMars — APPROVED** [review]
LGTM

**Review by MothOnMars — APPROVED** [review]
LGTM

**Pull request merged**